pkg/skaffold/access/provider.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2021 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package access
+
+import (
+	"sync"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/label"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubectl"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubernetes"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubernetes/portforward"
+)
+
+type Provider interface {
+	GetKubernetesAccessor(*kubernetes.ImageList) Accessor
+	GetNoopAccessor() Accessor
+}
```

Comment:
This is a little bit confusing.  Are all providers supposed to provide a no-op and kubernetes accessor?

Do we need a provider here?

Reply:
all provider should provide an implementation for each "style" of deployer we support in skaffold - right now there are only Kubernetes deployers (and they all share the same underlying logging/debugging/accessing implementations), but in the future we'll have alternate implementations.

I like this approach because it makes it clear what the different implementations for each component are, and also makes it clear where they can be obtained from (since they rely on dependencies that are injected via closures on the constructors). definitely open to suggestions for alternative approaches here though.